### PR TITLE
Add implementation support for selecting an existing vnet and subnet

### PR DESCRIPTION
**Pull request by milismsft**
Add implementation support for selecting an existing virtual network and subnet when creating a load balancer (ApplicationGateway) or server group.

### CODE REVIEW

**Comment by milismsft** [comment]
We did couple rounds of CR in the Microsoft fork for these changes before sending the public PR and looks good from our side.
@spinnaker/netflix-reviewers @duftler PTAL and merge if looks good for you as well.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy [hunk 1]
```diff
@@ -734,7 +734,12 @@ class AzureNetworkClient extends AzureBaseClient {
             }
 
             def vnet = AzureVirtualNetworkDescription.getDescriptionForVirtualNetwork(item)
-            vnet.subnets = AzureSubnetDescription.getSubnetsForVirtualNetwork(item, currentTime)
+            vnet.subnets = AzureSubnetDescription.getSubnetsForVirtualNetwork(item)
+
+            //
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy]: Extraneous comment.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/ops/UpsertAzureAppGatewayAtomicOperation.groovy [hunk 2]
```diff
@@ -63,6 +63,7 @@ class UpsertAzureAppGatewayAtomicOperation implements AtomicOperation<Map> {
     String resourceGroupName = null
     String virtualNetworkName = null
     String subnetName = null
+    String subnetId
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/ops/UpsertAzureAppGatewayAtomicOperation.groovy]: This guy's neighbors are all explicitly initialized to `null`.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy [hunk 3]
```diff
@@ -152,6 +156,14 @@ class AzureServerGroupDescription extends AzureResourceOpsDescription implements
     azureSG.appGatewayBapId = scaleSet.tags?.appGatewayBapId
     // TODO: appGatewayBapId can be retrieved via scaleSet->networkProfile->networkInterfaceConfigurations->ipConfigurations->ApplicationGatewayBackendAddressPools
     azureSG.subnetId = scaleSet.tags?.subnetId
+    azureSG.subnet = AzureUtilities.getNameFromResourceId(azureSG.subnetId)
+    if (azureSG.subnetId) {
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy]: This block would be more readable as:
```
azureSG.vnet = azureSG.subnetId ? azureSG.subnet : scaleSet.tags?.vnet
```

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/subnet/model/AzureSubnetDescription.groovy [hunk 4]
```diff
@@ -41,17 +47,55 @@ class AzureSubnetDescription extends AzureResourceOpsDescription {
...
 
-  static Collection<AzureSubnetDescription> getSubnetsForVirtualNetwork(VirtualNetwork vnet, long currentTime = 0) {
+  static Collection<AzureSubnetDescription> getSubnetsForVirtualNetwork(VirtualNetwork vnet) {
     // sort the list of subnet based on their ivp4 vals in order to speed the search when computing the next subnet
     vnet.subnets?.collect {
       getDescriptionForAzureSubnet(vnet, it)
     }?.sort { a,b -> a.ipv4 <=> b.ipv4}
   }
+
+  // This is API is an workaround which retrieves the Azure ApplicationGateway associations until applicationGatewayIPConfigurations is captured in the Azure Subnet
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/subnet/model/AzureSubnetDescription.groovy]: s/This is API is an/This API is a

**Comment by duftler** [comment]
LGTM once comments are address (and needs a rebase).

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy (hunk 3 above)
Comment (milismsft) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy]: It makes sense.

**Comment by milismsft** [comment]
@duftler i've made the suggested changes,;it will be awesome to get this checked in soon since the respective deck changes were already accepted and merged in.

**Comment by duftler** [comment]
Sorry, needs a rebase again.

**Pull request closed without merging**